solr/core/src/java/org/apache/solr/util/circuitbreaker/CircuitBreakerRegistry.java

```diff
@@ -111,43 +199,82 @@ public static String toErrorMessage(List<CircuitBreaker> circuitBreakerList) {
+    return circuitBreakerMap.containsKey(requestType)
+        || globalCircuitBreakerMap.containsKey(requestType);
   }
 
   @Override
   public void close() throws IOException {
     synchronized (circuitBreakerMap) {
-      final AtomicInteger closeFailedCounter = new AtomicInteger(0);
-      circuitBreakerMap
-          .values()
-          .forEach(
-              list ->
-                  list.forEach(
-                      it -> {
-                        try {
-                          if (log.isDebugEnabled()) {
-                            log.debug(
-                                "Closed circuit breaker {} for request type(s) {}",
-                                it.getClass().getSimpleName(),
-                                it.getRequestTypes());
-                          }
-                          it.close();
-                        } catch (IOException e) {
-                          if (log.isErrorEnabled()) {
-                            log.error(
-                                String.format(
-                                    Locale.ROOT,
-                                    "Failed to close circuit breaker %s",
-                                    it.getClass().getSimpleName()),
-                                e);
-                          }
-                          closeFailedCounter.incrementAndGet();
-                        }
-                      }));
+      closeCircuitBreakers(
+          circuitBreakerMap.values().stream().flatMap(List::stream).collect(Collectors.toList()));
       circuitBreakerMap.clear();
-      if (closeFailedCounter.get() > 0) {
-        throw new IOException("Failed to close " + closeFailedCounter.get() + " circuit breakers");
-      }
     }
   }
+
+  private static void closeGlobal() {
+    synchronized (globalCircuitBreakerMap) {
+      closeCircuitBreakers(
+          globalCircuitBreakerMap.values().stream()
+              .flatMap(List::stream)
+              .collect(Collectors.toList()));
+      globalCircuitBreakerMap.clear();
+    }
+  }
+
+  /**
+   * Close a list of circuit breakers, tracing any failures.
+   *
+   * @throws SolrException if any CB fails to close
+   */
+  private static void closeCircuitBreakers(List<CircuitBreaker> breakers) {
+    final AtomicInteger closeFailedCounter = new AtomicInteger(0);
+    breakers.forEach(
+        it -> {
+          try {
+            if (log.isDebugEnabled()) {
+              log.debug(
+                  "Closed circuit breaker {} for request type(s) {}",
+                  it.getClass().getSimpleName(),
+                  it.getRequestTypes());
+            }
+            it.close();
+          } catch (IOException e) {
+            if (log.isErrorEnabled()) {
+              log.error(
+                  String.format(
+                      Locale.ROOT,
+                      "Failed to close circuit breaker %s",
+                      it.getClass().getSimpleName()),
```

Comment:
In case of ambiguity e.g. a global search CB and a per-core update CB or so.
```suggestion
                      "Failed to close circuit breaker %s for request type(s) {}",
                      it.getClass().getSimpleName(),
                      it.getRequestTypes()),
```